Introduce config params for S3 timeouts.

mreid-moz opened the pull request:
This addresses #81

mreid-moz committed:
Introduce config params for S3 timeouts.

trink commented on heka/plugins/s3splitfile/s3splitfile_input.go:
```diff
@@ -64,6 +66,8 @@ func (input *S3SplitFileInput) ConfigStruct() interface{} {
 		S3ObjectMatchRegex: "",
 		S3Retries:          5,
+		S3ConnectTimeout:   120,
+		S3ReadTimeout:      120,
```
2 minutes seem excessive since it will block shutdown.  @whd opinions?

trink commented on heka/plugins/s3splitfile/s3splitfile_output.go:
```diff
@@ -114,19 +116,21 @@ const (
+		S3Bucket:         "",
+		S3BucketPrefix:   "",
+		S3Retries:        5,
+		S3ConnectTimeout: 300,
```
Why are the output defaults different?

mreid-moz replied:
Failures in S3Output are far more important than S3Input. We want to wait as long as possible before declaring failure. There is actual retry logic in the output (where that is not implemented in the input yet), so this should mostly be mitigated by retries, but tl;dr we want to try harder on Output.

mreid-moz committed:
Update default timeout to 60s.

Realistically, for any operation that will eventually succeed, it
should take far less than 60s.
